pulumi/README.md

```diff
@@ -0,0 +1,96 @@
+# @operator-error/pulumi-lambda-cert
+
+_This is a package containing components for use with [Pulumi][pulumi], a tool for provisioning cloud infrastructure 
+based on a description written with general purpose programming languages._
+
+This package provides a component named `LambdaCert`, which can be used to create the resources
+needed to run [Lambda Cert][lambdacert], a function for maintaining Let's Encrypt certificates using
+AWS Lambda.
```

Comment:
This is super cool that you can do this. One thing I've wondered from time to time, do you think we should eventually look into a Let's Encrypt resource provider, so that you can declare a certificate as a first class resource?

Reply:
This is more awkward. Terraform has an ACME provider, but I've personally found it to be of no use, as it introduces a runtime dependency to go and renew certificates. Since the validity period of certificates issued by Let's Encrypt is on the low side, this needs doing fairly often, hence the reason I built a Lambda to do it. This _may_ not be an issue for things with frequent iterations, but personally I like to have things "self-maintain" without requiring operator input.

The (missing) companion to this is a service which runs on instances making use of the certificates to download the certificate from the bucket each day, compare against the version the instance has, and reload the TLS configuration of the target service however necessary.

Using this is also only necessary when it's desirable to maintain TLS without interception all the way to the service instances - Vault is my canonical example. Most things running in AWS in particular can happily terminate TLS at the load balancer, API Gateway etc, and use ACM-issued certificates, which are strictly better.